### ISSUE TITLE
errorBorderColor & focusBorderColor couldn't be set using semantic tokens

### ISSUE THREAD

**Issue by theskinnycoder** [chakra-ui/chakra-ui#5653]
### Description

By using the semantic tokens feature introduced in v1.8+, I'm unable to set the focusBorderColor, or errorBorderColor properties to components like Input etc. I'm using Next.JS v12

### Link to Reproduction

https://codesandbox.io/s/compassionate-cray-e534ez?file=/src/index.js

### Steps to reproduce

1. **`_app.js`** : (Theme setup)

```js
import { extendTheme, ChakraProvider } from '@chakra-ui/react'

const theme = extendTheme({
  semanticTokens: {
    colors: {
	  primary: {
        default: "red.200",
      	_light: "red.500",
      },
  	},
  },
})

export default function MyApp({ Component, pageProps }) {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <Component {...pageProps} />
    </ChakraProvider>
  )
}
```

2. **`login.js`** : (Doesn't work)

```js

...

<Input
  type="text"
  focusBorderColor="primary"
/>

...

```

3. Tentative Solution :

```js

...

<Input
  type="text"
  focusBorderColor="red.200"
  _light={{
    _focus: {
	  borderColor: "red.500",
    },
  }}
/>

...

```

### Chakra UI Version

1.8.3

### Browser

Google Chrome v98.0.4

### Operating System

- [x] macOS
- [ ] Windows
- [ ] Linux

**Comment by m4x3d** [comment]
Hey, thanks for opening that issue!

I'm not sure if it is really a bug because the description of e.g. `focusBorderColor` says: 
*The border color when the input is focused. Use color keys in `theme.colors` @example focusBorderColor = "blue.500"*

So I guess it is not intended to be used with semantic tokens neither with some 'basic' CSS colors that names are used in the `theme.colors` like e.g. green.

Created a small example (on which I played a bit around as well :smile:)
https://codesandbox.io/s/upbeat-sid-jcde09?file=/src/index.tsx

Maybe @TimKolberger can verify it, saw more or less only his name within the `styled-system` package around the semantic token stuff as I looked around.

**Comment by TimKolberger** [comment]
## tldr;

It is currently not possible to use semantic tokens in `focusBorderColor` and `errorBorderColor`.
A workaround is to extend the theme, which is quite a lot code: [See CodeSandbox](https://codesandbox.io/s/semantic-tokens-for-input-focus-color-ni5vbx?file=/src/index.tsx)

## Long explanation

The theme styles for the Input component uses `getColor` to retrieve the color value from the theme:

https://github.com/chakra-ui/chakra-ui/blob/27eec8de744d05eef5bcbd2de651f3a37370ff2c/packages/theme/src/components/input.ts#L98-L106

The semantic tokens rely heavily on CSS variables, which can change with a CSS media query or selector like our dark mode selector.
Here is an example of the implementation detail:

```css
--chakra-colors-my-focus-color: #FF0000;
html[data-theme="dark"] {
  --chakra-colors-my-focus-color: #DD0000;  
}
```

Because of the ambiguity of semantic tokens it is not possible to retrieve an actual color value for it.

## Proposed Solution or API

The theme needs a new mechanism to retrieve theme tokens which should replace the usage of `getColor()` in the theme object.

```tsx
const reference = getToken(theme, token, fallback)
//  => var(--chakra-<token>)
```

This way we would be able to reference every token and semantic token in the theme, e.g.:

```ts
 _invalid: { 
   borderColor: getToken(theme, `colors.${ec}`, ec), 
   boxShadow: `0 0 0 1px ${getToken(theme, `colors.${ec}`, ec)}`, 
 }, 
 _focus: { 
   zIndex: 1, 
   borderColor: getToken(theme, `colors.${fc}`, fc), 
   boxShadow: `0 0 0 1px ${getToken(theme, `colors.${fc}`, fc)}`, 
 }, 
```

**Comment by TimKolberger** [comment]
Here is an implementation for the proposed `getToken` function: [CodeSandbox](https://codesandbox.io/s/semantic-tokens-for-input-focus-color-proposed-solution-lfykff)

**Comment by m4x3d** [comment]
Very nice! Thank you for the detailed explanation!

**Comment by theskinnycoder** [comment]
Found this work around : 

```js
const theme = extendTheme({
   components: {
     Input: {
       defaultProps: {
         focusBorderColor: 'brand.500',
       },
     },
   },
})
```
But do I alter it based on the color mode?

**Comment by anapaulalemos** [comment]
Hello guys, sorry for changing the subject a little bit!

@TimKolberger  I'm having problems trying to use a gradient as a theme token, do you know if the reason is related to the 'problem' with border colors?
I tried to set this gradient as a theme color and also as a semantic token and didn't have success.

I Created a small example:
https://codesandbox.io/s/dark-currying-w7kpb6?file=/src/App.tsx

Maybe I'm doing something wrong...

**Comment by anubra266** [comment]
> > @anapaulalemos You should be doing this:
> > ```tsx
> > import { Box, ChakraProvider, extendTheme } from "@chakra-ui/react";
> > import "./styles.css";
> > 
> > const theme = extendTheme({
> >   colors: {
> >     brand: {
> >       100: "#FF6677",
> >       200: "linear-gradient(90deg, #FFF53B 0%, #84dDA8 100%)"
> >     }
> >   },
> >   semanticTokens: {
> >     colors: {
> >       background: { default: "#FFF53B" },
> >       gradient: {
> >         default: "linear-gradient(90deg, #FFF53B 0%, #84dDA8 100%)"
> >       }
> >     }
> >   }
> > });
> > 
> > export default function App() {
> >   return (
> >     <ChakraProvider theme={theme}>
> >       <Box bgColor="brand.100">
> >         <h1>Hello using color</h1>
> >       </Box>
> > 
> >       <Box bgColor="background">
> >         <h1>Hello using semantic token color</h1>
> >       </Box>
> > 
> >       <Box bg="gradient">
> >         <h1>Hello using semantic token gradient</h1>
> >       </Box>
> > 
> >       <Box bgGradient="linear(90deg, #FFF53B 0%, #84dDA8 100%)">
> >         <h1>Hello using gradient</h1>
> >       </Box>
> >     </ChakraProvider>
> >   );
> > }
> > ```
> 
> Hey @anubra266, how are you? Thank you so much for your help! It works like a charm! :)

I'm fine thanks. Glad it worked

**Comment by anapaulalemos** [comment]
Hey @anubra266, do you know how to make a semantic token gradient work using a chakra color instead of a raw CSS color value?

E.g:

```tsx
import { Box, ChakraProvider, extendTheme } from "@chakra-ui/react";
import "./styles.css";

const theme = extendTheme({
  semanticTokens: {
    colors: {
      background: { default: "#FFF53B" },
      gradient: {
        default: "linear-gradient(90deg, #FF6677 0%, #84dDA8 100%)"
      },
      otherGradient: {
        default: "linear-gradient(90deg, red.500 0%, red.200 100%)"
      }
    }
  }
});

export default function App() {
  return (
    <ChakraProvider theme={theme}>

      {/* this works */}
      <Box bg="gradient">
        <h1>Hello using semantic token gradient</h1>
      </Box>

      {/* this don't work */}
      <Box bg="otherGradient">
        <h1>Hello using semantic token gradient with colors</h1>
      </Box>
    </ChakraProvider>
  );
}

```  

Thanks in advance!

**Comment by anubra266** [comment]
Hey @anapaulalemos I would use CSS tokens. like this
```js
const theme = extendTheme({
  semanticTokens: {
    colors: {
      gradient: {
        default: "linear-gradient(90deg, var(--chakra-colors-red-500) 0%, var(--chakra-colors-red-200) 100%)"
      }
    }
  }
});
```
